feat: Implement Dispositif Origin Field (RI-977)

kaaloo opened the pull request:
## Summary
Implements the `origin` field for dispositifs to distinguish between "RI" (Refugies.info) and "RCO" (Reseau Carif-Oref) content. This includes backend changes, frontend adaptations, and a database migration.

## Description
This PR implements the "Dispositif Origin Metadata" feature (RI-977).

### Changes
- **Backend**:
    - Added `origin` field to `Dispositif` schema and types (`DispositifResponse`, `SimpleDispositif`).
    - Updated `createDispositif` and `updateDispositif` workflows.
    - Added database migration to set default `origin` to "RI".
- **Frontend**:
    - Updated `Dispositif` components to handle "RCO" origin.
    - Added placeholder page for "RCO" dispositifs.
    - Disabled editing features for "RCO" dispositifs.
- **Testing**:
    - Added unit tests for `origin` field logic.
    - Verified "RI" and "RCO" behavior in the webapp.

### Artifacts
- `specs/003-dispositif-origin-meta/spec.md`: Functional requirements and user stories.
- `specs/003-dispositif-origin-meta/plan.md`: Implementation plan and technical context.
- `specs/003-dispositif-origin-meta/data-model.md`: Schema design (Enum, Immutable).
- `specs/003-dispositif-origin-meta/contracts/`: API contract updates.
- `specs/003-dispositif-origin-meta/tasks.md`: Detailed task breakdown.

## Status
- [x] Specification Complete
- [x] Plan Complete
- [x] Tasks Generated
- [x] Implementation Complete
- [x] Tests Passed

kaaloo committed:
docs: add specification for dispositif origin metadata tracking

- Created feature spec for distinguishing editorial (RI) vs AI-assisted (RCO) dispositif sources
- Added requirements checklist template for specification quality validation
- Defined functional requirements for origin field persistence, validation, and API exposure
- Outlined user scenarios for frontend integration and backward compatibility
- Specified success criteria for API response completeness and analytics accuracy

kaaloo committed:
docs(dispositif): add implementation docs for origin

Add planning, data model, API contracts, research, and
quickstart documentation for the dispositif origin metadata
feature (RI-977).

kaaloo committed:
docs(dispositif): add tasks for origin meta spec

kaaloo committed:
chore: :sparkles: Finish up spec analysis

ledjay commented on specs/003-dispositif-origin-meta/research.md:
```diff
@@ -0,0 +1,171 @@
+1. Add `origin` field to Mongoose schema with `enum: ["RI", "RCO"]` and `default: "RI"`
+2. Existing documents without `origin` will use the default when queried
+3. New/updated records will always have `origin` explicitly set
+4. Optional future migration script can backfill if needed for analytics/reporting
```
I know the bulk option would be heavy and risky but on the other hand I'm afraid of data inconsistencies and bugs if we don't specify the origin RI right away.

WDYT @kaaloo ?

kaaloo replied:
Thanks, yes I added a task to create the migration for this, nice catch !

ledjay commented on specs/003-dispositif-origin-meta/data-model.md:
```diff
@@ -0,0 +1,263 @@
+  // ... existing fields ...
+
+  @prop({
+    enum: ["RI", "RCO"],
```
Quick idea: maybe it's premature and enums will to the trick but it seems future proof to handle the origins from a separated mongo table to handle that kind of data later on :
{
handle : "RCO",
name : {
fr: "Réseau Carif-Oref"
},
description : {
fr: "blabla"
},
url: "https://....",
}
WDYT ?

kaaloo replied:
Definitely, but we should discuss future proofing.  In general I prioritize meeting short term objectives and simplicity over getting a general design right in the early stages.  This is of course conditioned to having agile refactoring capabilities in place.

ledjay commented on specs/003-dispositif-origin-meta/data-model.md:
```diff
@@ -0,0 +1,263 @@
+2. No bulk update operation needed
+3. Optional future backfill script can be created if needed for analytics
+
+### Optional Future Backfill (if needed)
```
I vote for this right away 🗳️ :)

kaaloo committed:
chore(dispositif): add migration tasks for origin field backfill

- Added task T004a to generate migration file for backfilling origin field
- Added task T004b to implement migration logic setting origin: "RI" on existing dispositifs

kaaloo committed:
chore: Code review updates

ledjay reviewed (approved):
LGTM !

kaaloo committed:
chore(turbo): exclude AI assistant directories from build inputs

- Added .kilocode, .specify, and .windsurf directories to input exclusions across all tasks
- Prevents unnecessary cache invalidation from AI assistant metadata changes
- Applied formatting consistency to globalDependencies array

kaaloo committed:
chore: upgrade spec kit to 0.0.22 and latest templates.  Add support for Kilo Code.

kaaloo committed:
docs(dispositif): refine scope and add mobile tasks for origin feature

- Clarified FR-005 to exclude backoffice UI updates from current scope
- Added note to T010 that RCO content is French-only (no translation keys needed)
- Expanded T016 mobile placeholder into concrete subtasks:
  - T016: Update mobile Dispositif type definition
  - T016a: Create OriginBadge component with DSFR-adapted styles
  - T016b: Add badge to search result items
  - T016c: Add badge to detail screen

kaaloo committed:
docs(dispositif): mark requirements checklist as complete

kaaloo committed:
chore: add standard ignore patterns to eslint and prettier configs

- Added node_modules, dist, build, and coverage directories to both ignore files
- Added minified JS files to eslintignore
- Added lock files (package-lock.json, yarn.lock, pnpm-lock.yaml) to prettierignore

kaaloo committed:
chore: sync specify rules across AI assistants and add MongoDB to tech stack

- Created .kilocode/rules/specify-rules.md with current project guidelines
- Added MongoDB (existing dispositifs collection) to active technologies in windsurf rules
- Both files now track same tech stack and conventions for consistency

kaaloo committed:
feat(dispositif): add migration to backfill origin field with "RI" value

- Created migration to set origin: "RI" on all existing dispositifs without an origin field
- Added rollback logic to remove origin field from dispositifs with "RI" value
- Logs count of modified documents during migration

kaaloo committed:
feat(dispositif): add origin field UI display and filtering across all platforms

- Added origin badge display on dispositif cards showing "Généré par IA" for RCO content
- Added DSFR pink tuile color scheme for RCO badges (web and mobile)
- Added RCO disclaimer callout on dispositif detail pages (web and mobile)
- Added filterByOrigin function for search filtering by content origin
- Added origin field to Algolia indexing format
- Added defensive array checks for secondaryThemes and needs in

kaaloo committed:
chore: remove Kilo Code assistant configuration files

- Removed .kilocode/rules/specify-rules.md with project guidelines
- Removed .kilocode/workflows/ directory containing speckit.analyze.md, speckit.checklist.md, and speckit.implement.md workflow definitions

kaaloo committed:
chore: get rid of kilo code spec kit integration and add gemini cli

- Updated .eslintignore and .prettierignore to exclude .gemini directory instead of .kilocode
- Updated turbo.json input exclusions across all tasks to use .gemini directory
- Maintains consistent AI assistant directory exclusion pattern

kaaloo committed:
chore: add support for opencode in Spec Kit

kaaloo committed:
chore: remove OpenCode assistant configuration files

- Removed .opencode/command/ directory containing speckit.analyze.md, speckit.checklist.md, speckit.clarify.md, and speckit.implement.md workflow definitions

kaaloo committed:
refactor: replace `require` with `import` for `getDispositifById` and remove `@ts-ignore` comments.

kaaloo committed:
feat: initialize dispositif origin and simplify fixture data by removing content, advancement, and audience age fields.

kaaloo commented:
/gemini review
